protocol/messenger_handler.go

```diff
@@ -583,6 +583,8 @@ func (m *Messenger) handleSyncChats(messageState *ReceivedMessageState, chats []
 func (m *Messenger) HandleSyncInstallationContactV2(state *ReceivedMessageState, message *protobuf.SyncInstallationContactV2, statusMessage *v1protocol.StatusMessage) error {
 	// Ignore own contact installation
 
+	fromBackup := statusMessage == nil
```

Comment:
It's not clear why having no message means it comes from backups :thinking:

Reply:
Well, there are 3 places where this function is called from:
1. `handleBackup` (used when applying messages from the backup)
2. `HandleSyncRawMessages` (used when local pairing)
3. `dispatchToHandler` -> `handleSyncInstallationContactV2Protobuf` (used when a synced message comes from another device over the network)

In case of point 3 the code is the same as it was, in case of points 1 and 2 we need to store the contact even it has `Removed` or `Blocked` flag set to `false`.

So the issue was spotted in https://github.com/status-im/status-mobile/issues/18096 but based on the code the same happens for Removed or Blocked contacts for applying messages from backup and local pairing. That should be fixed now.